addons/binding/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/util/UidUtils.java

```diff
@@ -32,9 +36,31 @@ public static ThingUID generateThingUID(Device device, Bridge account) {
     }
 
     /**
+     * Returns all ThingUIDs for a given device.
+     */
+    public static List<ThingUID> getThingUIDs(Device device, Bridge account) {
+        List<ThingUID> thingUIDs = new ArrayList<ThingUID>();
+        for (Thing thing : account.getThings()) {
+            String deviceId = thing.getConfiguration().as(GardenaDeviceConfig.class).deviceId;
+            if (deviceId == null) {
+                deviceId = thing.getUID().getId();
+            }
+            if (device.getId().equals(deviceId)) {
```

Comment:
You just guaranteed that the `deviceId` is not `null`, maybe switch the equals around?